phoenix-core/src/main/java/org/apache/phoenix/iterate/BaseResultIterators.java

```diff
@@ -550,6 +550,76 @@ public BaseResultIterators(QueryPlan plan, Integer perScanLimit, Integer offset,
         }
         return ranges;
     }
+
+    private List<KeyRange> getRegionRowKeyRanges() throws SQLException {
+        List<HRegionLocation> regionLocations = getRegionBoundaries(scanGrouper);  //Load the region information
+
+        List<KeyRange> regionKeyRanges = Lists.newArrayListWithExpectedSize(regionLocations.size());
+
+        //Map each HRegionLocation to a KeyRange - no Java 8
+        for(HRegionLocation regionLocation : regionLocations){
+            HRegionInfo regionInfo = regionLocation.getRegionInfo();
+
+            //Region is upperInclusive false by definition
+            KeyRange range = KeyRange.getKeyRange(regionInfo.getStartKey(), true, regionInfo.getEndKey(), false);
+            regionKeyRanges.add(range);
+        }
+        return regionKeyRanges;
+    }
+
+    List<List<KeyRange>> getQueryRowKeyRanges(List<KeyRange> regionKeyRanges) {
+        ScanRanges scanRanges = context.getScanRanges();
+
+        RowKeySchema schema = scanRanges.getSchema();
+
+        PTable table = getTable();
+        boolean isLocalIndex = table.getIndexType() == IndexType.LOCAL;
+
+        //Use the dataplan to build the queryRowKeyRanges
+        if (isLocalIndex) {
+            // should always have a data plan when a local index is being used.
+            if (dataPlan != null && dataPlan.getTableRef().getTable().getType()
+                    != PTableType.INDEX) { // Sanity check
+                scanRanges =
+                        computePrefixScanRanges(dataPlan.getContext().getScanRanges(),
+                                computeColumnsInCommon());
```

Comment:
I prefer the following coding style which seems to be used by phoenix code base generally.

 <type >scanRanges = computePrefixScanRanges(
              dataPlan.getContext().getScanRanges(), computeColumnsInCommon());

Did you follow the following instructions on https://phoenix.apache.org/develop.html to set phoenix coding style.
Get Settings and Preferences Correct
Import Settings from eclipse profile
File -> Settings -> Editor -> Code Style -> Java
Set From… -> Import… -> Eclipse XML Profile -> {repository}/dev/PhoenixCodeTemplate.xml

Reply:
I thought this was already on but appeared this was a new workspace.  I have applied in general.